src/controller.rs

```diff
@@ -302,7 +303,17 @@ async fn delete_cleanup_hack_configmap(
     let delete_params = DeleteParams::default();
     let cm_name = &delete::cleanup_hack_resource_name(app_instance);
     info!("Performing ConfigMap deletion on {cm_name} to finalise cleanup process.");
-    cm_api.delete(cm_name, &delete_params).await?;
+    cm_api
+        .delete(cm_name, &delete_params)
+        .await
+        .map(|_| ())
+        .or_else(|err| match err {
+            // ConfigMap has already been deleted or does not exist, so there
+            // is nothing to delete.
+            kube::Error::Api(ErrorResponse { code: 404, .. }) => Ok(()),
+            _ => Err(err),
+        })
+        .map_err(Error::KubeError)?;
```

Comment:
Commenting so that this stealth commit doesn't go unnoticed in case we add the `automerge` on this.

As part of some testing I noticed that if the `ConfigMap` doesn't get created/doesn't exist (e.g. if the `imagePullSecrets` aren't valid) and you want to get rid of your `AppInstance` anyway, this would've previously continually errored out, but a `404` on this deletion is okay so that it doesn't block indefinitely.

I nabbed this trick from https://github.com/kube-rs/kube/blob/44932971fe12ddeebde4a85bb92b8a2fb7f0e5e3/examples/secret_syncer.rs#L63-L72